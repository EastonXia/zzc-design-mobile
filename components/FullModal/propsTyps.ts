
export interface FullModalProps {
    prefixCls?: string
    className?: string
    title?: string
    visible: boolean
    closeCallback: Function
    BScrollInitCallback: Function
    style?: React.CSSProperties
    isUseBScroll?: boolean
    BScrollOpt?: Object
}