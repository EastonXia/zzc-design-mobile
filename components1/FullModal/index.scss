@import '../style/base.scss';
$cls: '.zds-full-modal';
#{$cls} {
    &-box {
        height: 100%;
        width: 100%;
        background: $c_fff;
        #{$cls}-content {
            height: 100%;
            width: 100%;
            display: flex;
            flex-direction: column;
            #{$cls}-head {
                padding: 64px 15px 9px;
                position: relative;
                &::after {
                    position: absolute;
                    content: '';
                    display: block;
                    z-index: 11;
                    bottom: 0;
                    left: 50%;
                    width: 23.5px;
                    border-bottom: 4px solid $main;
                    transform: translateX(-50%);
                }
                h5 {
                    color: #282828;
                    padding: 0;
                    font-size: 22px;
                    line-height: 1.5;
                    text-align: center;
                }
            }
            #{$cls}-body {
                flex: 1;
                margin-top: 23px;
                height: 400px;
                padding: 0 15px;
                font-size: $fs_menu;
                overflow-y: scroll;
                overflow-x: hidden;
                word-wrap: break-word;
                -webkit-overflow-scrolling: touch;
                &#{$cls}-wrapper{
                    overflow-y: hidden;
                }
            }
            #{$cls}-footer {
                margin: 25px 0 40px;
                display: flex;
                justify-content: center;
                position: relative;
                &::after {
                    display: block;
                    content: '';
                    position: absolute;
                    top: -22.5px;
                    transform: translateY(-100%);
                    left: 0;
                    width: 100%;
                    height: 20px;
                    background: linear-gradient(rgba(255, 255, 255, .1), rgba(255, 255, 255, 1));
                    ;
                    z-index: 999;
                }
                #{$cls}-btn {
                    display: flex;
                    justify-content: center;
                    align-items: center;
                    width: 46px;
                    height: 46px;
                    border-radius: 50%;
                    border: 1px solid $c_666;
                    &-active {
                        background: #e1e1e1;
                    }
                    span {
                        width: 46px;
                        height: 46px;
                        position: relative;
                        transform: rotate(45deg);
                        &::after {
                            position: absolute;
                            top: 50%;
                            left: 50%;
                            transform: translateX(-50%) translateY(-50%);
                            display: block;
                            content: '';
                            width: 50%;
                            height: 1px;
                            background: $c_666;
                        }
                        &::before {
                            position: absolute;
                            top: 50%;
                            left: 50%;
                            transform: translateX(-50%) translateY(-50%) rotate(90deg);
                            display: block;
                            content: '';
                            width: 50%;
                            height: 1px;
                            background: $c_666;
                        }
                    }
                }
            }
        }
    }
}