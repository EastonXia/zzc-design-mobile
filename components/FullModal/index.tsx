import React, { PureComponent } from 'react';
import Error from '../_util/Error';
import FullModal from './components/FullModal';
import { FullModalProps } from './propsTyps';
import './index.scss';

export default class FullModalEntrance extends PureComponent<FullModalProps, any> {
    render() {
        return (
            <Error componentName='FullModal'>
                <FullModal {...this.props} />
            </Error>
        );
    }
}