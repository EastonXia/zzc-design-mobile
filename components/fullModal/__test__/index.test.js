import React from 'react';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import FullModal from '../index.tsx';

test( 'Modal', () => {
    const wrapper = render( <div>
        <FullModal
            visible
            title='这是一个标题'
            closeCallback={() => {}}
        >
            <div className='modal-text'>这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案</div>
        </FullModal>
        <FullModal
            visible
            title='这是一个标题'
            closeCallback={() => {}}
            isUseBScroll
        >
            <div className='modal-text'>这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案</div>
        </FullModal>
    </div> );
    expect( renderToJson( wrapper ) ).toMatchSnapshot();
} );

