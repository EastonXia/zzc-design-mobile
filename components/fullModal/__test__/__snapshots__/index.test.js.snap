// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 1`] = `
<div>
  <div
    class="zds-dialog"
  >
    <div
      class="zds-dialog-box zds-full-modal-box"
    >
      <div
        class="zds-full-modal-content"
      >
        <div
          class="zds-full-modal-head"
        >
          <h5>
            这是一个标题
          </h5>
        </div>
        <div
          class="zds-full-modal-body"
        >
          <div
            class="modal-text"
          >
            这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案
          </div>
        </div>
        <div
          class="zds-full-modal-footer"
        >
          <div
            class="zds-full-modal-btn"
          >
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="zds-dialog"
  >
    <div
      class="zds-dialog-box zds-full-modal-box"
    >
      <div
        class="zds-full-modal-content"
      >
        <div
          class="zds-full-modal-head"
        >
          <h5>
            这是一个标题
          </h5>
        </div>
        <div
          class="zds-full-modal-body zds-full-modal-wrapper"
        >
          <div
            class="zds-full-modal-wrapper-content"
          >
            <div
              class="modal-text"
            >
              这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案，这里是说明文案
            </div>
          </div>
        </div>
        <div
          class="zds-full-modal-footer"
        >
          <div
            class="zds-full-modal-btn"
          >
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
